pkg/traceql/engine_metrics.go

```diff
@@ -876,9 +993,13 @@ func (e *MetricsEvalulator) Do(ctx context.Context, f SpansetFetcher, fetcherSta
 			}
 
 			e.spansTotal++
-			e.metricsPipeline.observe(s)
 
+			if e.sampleExemplar(ss.TraceID) {
+				e.metricsPipeline.observe(s, ss.TraceID)
```

Comment:
>(1) the current code as-is needs to at least add a continue statement, right now it is observing the span twice.

Nice catch. Fixed!

>(2) But since we are only sampling once per traceID, the sampling decision can go outside the loop entirely. We could always sample span 0, or even choose a random one (since we know the length of ss.Spans). I'm fine with just sampling span 0 to start.

Good point. It should save some cycles and makes it a bit cleaner. Changed!